src/php/Config/ProjectConfiguration.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Phel\Config;
+
+use JsonSerializable;
+use Phel\Command\CommandConfig;
+use Phel\Interop\InteropConfig;
+
+final class ProjectConfiguration implements JsonSerializable
+{
+    private TestConfiguration $testConfig;
+
+    private ExportConfiguration $exportConfig;
+
+    public function __construct()
+    {
+        $this->testConfig = TestConfiguration::empty();
+        $this->exportConfig = ExportConfiguration::empty();
+    }
+
+    public function setTestConfiguration(TestConfiguration $testConfig): self
+    {
+        $this->testConfig = $testConfig;
+
+        return $this;
+    }
+
+    public function setExportConfiguration(ExportConfiguration $exportConfig): self
+    {
+        $this->exportConfig = $exportConfig;
+
+        return $this;
+    }
+
+    public function jsonSerialize(): array
+    {
+        return [
+            CommandConfig::TESTS => $this->testConfig->getDirectories(),
+            InteropConfig::EXPORT => [
+                InteropConfig::EXPORT_DIRECTORIES => $this->exportConfig->getDirectories(),
+                InteropConfig::EXPORT_NAMESPACE_PREFIX => $this->exportConfig->getNamespacePrefix(),
+                InteropConfig::EXPORT_TARGET_DIRECTORY => $this->exportConfig->getTargetDirectory(),
+            ],
+        ];
+    }
```

Comment:
I would prefer to create a separate follow-up PR to apply this change because this will break the current BC:
```php
return [
    'tests_directories' => $this->testsDirectories,
    'export_directories' => $this->exportConfig->getDirectories(),
    'export_namespace_prefix' => $this->exportConfig->getNamespacePrefix(),
    'export_target_directory' => $this->exportConfig->getTargetDirectory(),
];
```
> TL;DR: I want to keep the BC in this PR.